Source/ROLib/Modules/ROLAnimateEngineHeat.cs

```diff
@@ -0,0 +1,216 @@
+﻿using System;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace ROLib
+{
+    public class ROLAnimateEngineHeat : PartModule
+    {
+
+        //amount of 'heat' added per second at full throttle
+        [KSPField]
+        public float heatOutput = 300;
+
+        //amount of heat dissipated per second, adjusted by the heatDissipationCurve below
+        [KSPField]
+        public float heatDissipation = 100;
+
+        //point at which the object will begin to glow
+        [KSPField]
+        public float draperPoint = 400;
+
+        //maximum amount of heat allowed in this engine
+        //will reach max glow at this temp, and begin dissipating even faster past this point
+        [KSPField]
+        public float maxHeat = 2400;
+
+        //maxStoredHeat
+        //storedHeat will not go beyond this, sets retention period for maximum glow
+        [KSPField]
+        public float maxStoredHeat = 3600;
+
+        //curve to adjust heat dissipation; should generally expel heat faster when hotter
+        [KSPField]
+        public FloatCurve heatDissipationCurve = new FloatCurve();
+
+        //the heat-output curve for an engine (varies with thrust/throttle), in case it is not linear
+        [KSPField]
+        public FloatCurve heatAccumulationCurve = new FloatCurve();
+
+        [KSPField]
+        public FloatCurve redCurve = new FloatCurve();
+
+        [KSPField]
+        public FloatCurve blueCurve = new FloatCurve();
+
+        [KSPField]
+        public FloatCurve greenCurve = new FloatCurve();
+
+        [KSPField]
+        public string engineID = "Engine";
+
+        [KSPField]
+        public String meshName = String.Empty;
+
+        [KSPField(isPersistant = true)]
+        public float currentHeat = 0;
+
+        [KSPField]
+        public bool useThrottle;
+
+        int shaderEmissiveID;
+
+        private ModuleEngines engineModule;
+
+        private Renderer[] animatedRenderers;
+
+        private Color emissiveColor = new Color(0f, 0f, 0f, 1f);
+
+        public override void OnAwake()
+        {
+            base.OnAwake();
```

Comment:
No purpose to calling base.On[*]() in PartModules.  The On* calls originate in the PartModule class.